### PR TITLE
 Support `valid` padding in Conv2D

### PR DESCRIPTION
**Pull request by AdamHillier**
For some work I was doing recently I needed to verify networks that had convolutional layers with `valid` padding, and so I made some light modifications to support such layers in MIPVerify. What do you think of this approach?

### CODE REVIEW

**Comment by vtjeng** [comment]
Thanks for the PR! Looks like a good approach overall 😄 I'm going to take a closer look as soon as I can --- hopefully within the week --- and leave some more specific comments.

**Comment by IsSeck** [comment]
Hello, 

I apologize if this is not the place for my question. 

I wanted to know if padding=valid is implemented and if it is how to use it with get_conv_params if possible? Thank you

**Comment by vtjeng** [comment]
Thank you --- this is the right place for your question.

Unfortunately, I haven't had the chance to merge `padding=valid` in just yet; it conflicts partially with a refactor that I merged. You should be able to modify the source code of the package on disk to implement this, and you may be able to try to merge in the changes in this PR (although you'll have to resolve the merge conflict).

**Comment by samuelemarro** [comment]
Valid padding support would be really helpful. If you need help with the pull request, I'd be glad to help!

**Comment by vtjeng** [comment]
@samuelemarro, that'll be much appreciated. You should be able to use much of Adam's code (feel free to open a new pull request, we'll make sure that Adam is properly credited). It would be best if you could add testing as mentioned above. https://github.com/vtjeng/MIPVerify.jl/pull/16#pullrequestreview-254066901

(Also, if you are planning to use this for your own networks, a sanity check on real values --- e.g. forward propagating input through the network and making sure it matches what you get in Pytorch / Tensorflow --- would be another great test to make sure we are not getting the convention subtly wrong).

**Comment by AdamHillier** [comment]
Hi all, I'm very sorry that I never got round to coming back to this. @samuelemarro please be my guest to move forward with this code! I'll close this for now and you can open a new PR as @vtjeng suggests.
